src/app/project/manage-payments/manage-payments.component.ts

```diff
@@ -14,43 +14,58 @@ import { WalletDetails } from '../../shared/services/wallet/wallet-cooperative/w
     styleUrls: ['./manage-payments.component.css']
 })
 export class ManagePaymentsComponent implements OnInit {
+
+    @Input() revenueShareAmount;
     projectWallet: WalletDetails;
     project: Project;
 
     constructor(private walletService: WalletService,
                 private route: ActivatedRoute,
+                private router: Router,
                 private projectService: ProjectService) {
     }
 
     ngOnInit() {
-
         const projID = this.route.snapshot.params.projectID;
-
         this.getProjectWallet(projID);
         this.getProject(projID);
     }
 
     getProject(projectID: string) {
         SpinnerUtil.showSpinner();
-        this.projectService.getProject(projectID).subscribe((res: any) => {
-            SpinnerUtil.hideSpinner();
-            this.project = res;
-        }, err => {
-            SpinnerUtil.hideSpinner();
-            displayBackendError(err);
-        });
+        this.projectService.getProject(projectID)
+            .subscribe((res) => {
+                this.project = res;
+            }, err => {
+                displayBackendError(err);
+            })
+            .add(SpinnerUtil.hideSpinner);
     }
 
     getProjectWallet(projectID: number) {
         SpinnerUtil.showSpinner();
-        this.walletService.getProjectWallet(projectID).subscribe(res => {
-            SpinnerUtil.hideSpinner();
-            this.projectWallet = res;
-            this.projectWallet.currency = prettyCurrency(res.currency);
-            this.projectWallet.balance = numeral(res.balance).format('0,0');
-        }, err => {
-            SpinnerUtil.hideSpinner();
-            displayBackendError(err);
-        });
+        this.walletService.getProjectWallet(projectID)
+            .subscribe(res => {
+                this.projectWallet = res;
+                this.projectWallet.currency = prettyCurrency(res.currency);
+                this.projectWallet.balance = numeral(res.balance).format('0,0');
+                autonumericCurrency('#revenueShareAmount');
+            }, err => {
+                displayBackendError(err);
+            })
+            .add(SpinnerUtil.hideSpinner);
+    }
+
+    startPayout() {
+        const projID = this.route.snapshot.params.projectID;
+        const orgID = this.route.snapshot.params.groupID;
+
+        this.router.navigate([
+            `/dash/manage_groups/${orgID}/manage_project/${projID}/manage_payments/revenue_share/${this.revenueShareAmount}`]);
+    }
+
+    onRevenueShareAmountChange(revenueShareAmount) {
+        (<HTMLInputElement>document.getElementById('buttonStartPayout'))
+            .disabled = revenueShareAmount === undefined || (revenueShareAmount.replace( /\D+/g, '')) <= 0;
```

Comment:
Please use reactive forms for these complex situations. They are useful and easy to use. We will use them in many cases later, so it will be good to know.
For beginning, watch [this](https://www.youtube.com/watch?v=JeeUY6WaXiA) video.